browser/src/control/Control.NavigatorPanel.ts

```diff
@@ -419,42 +419,74 @@ class NavigatorPanel extends SidebarBase {
 		this.builder.build(wrapper, [data], false);
 	}
 
-	override callback(
+	useDefaultCallback(
 		objectType: string,
 		eventType: string,
 		object: any,
 		data: any,
 		builder: JSBuilder,
-	): void {
-		let searchTerm = '';
-		// Update results tab
-		if (object.id === 'navigator-search-button') {
-			searchTerm = (
-				document.getElementById('navigator-search-input') as HTMLInputElement
-			).value;
-			super.callback('edit', 'activate', { id: 'Find' }, searchTerm, builder);
-		} else if (object.id === 'navigator-search') {
-			searchTerm = data;
-			super.callback(
-				objectType,
-				eventType,
-				{ id: 'Find' },
-				searchTerm,
-				builder,
-			);
+	) {
+		super.callback(objectType, eventType, object, data, builder);
+	}
+
+	useSearchCallback(
+		objectType: string,
+		eventType: string,
+		object: any,
+		builder: JSBuilder,
+	) {
+		// Switch to "Results tab" first.
+		if (eventType === 'activate') {
+			const resultsTab = document.getElementById('tab-quick-find');
+			if (resultsTab && !resultsTab.classList.contains('selected'))
+				resultsTab.click();
 		}
+
+		const nextButtonContainer = document.getElementById('findnext');
```

Comment:
can we do not search in whole document? Instead I would prefer querySelector on container of the quickfind panel -> it will prevent us from hitting duplicated id (which shouldn't be duplicated too...)

"findnext" sounds very generic, also you can do both queries once:
`panel.querySelector('#findnext button')`

so there is only one line :)